product-service/src/getProductById.js

```diff
@@ -0,0 +1,20 @@
+import { errorResponse, successResponse } from "./utils/response.js";
+import { mockProductList } from "./mockData.js";
+
+export const getProductById = async (event) => {
+  try {
+
+    const { productId = '' } = event.pathParameters;
+
+    const product = mockProductList.find((p) => productId === p.id);
+
+    if( product )
+      return successResponse(product);
+
+
+    return successResponse( { message: "Product not found" }, 404 );
```

Comment:
Minor issue: SuccessResponse is returned here, but in fact, you are returning an error. It might be better to rename the "SuccessResponse" method (for example, formatJSONResponse) or use ErrorResponse